### ISSUE TITLE
Fixing variable name does not rename all usages

### ISSUE THREAD

**Issue by max-frai** [rust-analyzer/rust-analyzer#3633]
Convert variable name to snake case will rename it, but all usages will remain the same.

Before:
<img width="513" alt="Screenshot 2020-03-18 at 14 00 01" src="https://user-images.githubusercontent.com/24208746/76958714-e8926680-6920-11ea-93f4-d94197d6fbed.png">

After:
<img width="216" alt="Screenshot 2020-03-18 at 14 02 28" src="https://user-images.githubusercontent.com/24208746/76958851-1c6d8c00-6921-11ea-92ba-ae9f56c57b3c.png">

**Comment by bjorn3** [comment]
That warning is generated by rustc, not rust-analyzer.

**Comment by Veetaha** [comment]
I wonder where do those `rustc` errors come from. Isn't it from our cargo watcher?
Anyway, @edwin0cheng I think this has something to do with macros?

**Comment by Veetaha** [comment]
I also think this is related to #3622

**Comment by max-frai** [comment]
I think this is rust-analyzer helper. Also it does not work not only for macros
<img width="280" alt="Screenshot 2020-03-18 at 14 09 32" src="https://user-images.githubusercontent.com/24208746/76959351-188e3980-6922-11ea-8e47-5931d76b0757.png">

**Comment by edwin0cheng** [comment]
I don't think it is related to macro here: 
```
fn main() {
    let foo<|>Bar = 1234; 
    let _ = fooBar;  // <- it won't be converted 
}
```

And I think it is from `cargo check`, because the search result of  `snake case` is empty in RA project :)

**Comment by kiljacken** [comment]
Yeah it's a diagnostic from rustc, they typically aren't all that smart. The right thing to do would probably be to report it upstream. They do mark it `MaybeIncorrect` in applicablity so maybe we should only suggest things they mark as `MachineApplicable`.

**Comment by flodiebold** [comment]
Duplicate of #3391

**Comment by lnicola** [comment]
After #7508 we now show two different fixes, one from `rustc` and one of our own:

![image](https://user-images.githubusercontent.com/308347/106494171-c6ff8680-64c2-11eb-8b96-0c6c9ecbb9c9.png)
